images/upload-gitlab-failure-logs/upload_gitlab_failure_logs.py

```diff
@@ -63,25 +91,9 @@ def main():
     job = project.jobs.get(job_id)
     job_trace: str = job.trace().decode()  # type: ignore
 
-    job_error_class = None
-    matching_patterns = set()
-    for error_class, lookups in taxonomy["error_classes"].items():
-        if lookups:
-            for grep_expr in lookups.get("grep_for", []):
-                if re.compile(grep_expr).search(job_trace):
-                    matching_patterns.add(error_class)
-
-    # If the job logs matched any regexes, assign it the taxonomy
-    # with the highest priority in the "deconflict order".
-    # Otherwise, assign it a taxonomy of "other".
-    if len(matching_patterns):
-        for error_class in taxonomy["deconflict_order"]:
-            if error_class in matching_patterns:
-                job_error_class = error_class
-                break
-    else:
-        job_error_class = "other"
+    job_error_class = classify(job_trace)
```

Comment:
And here do something like:
```suggestion
    matches, job_error_class = classify(job_trace)
    job_input_data["matches"] = sorted(matches)
```
?

Reply:
I think one good reason is not filtering to a single match can lead to mislabeled classifications. The ordering below is meant to identify the "best' error match.

For example, `file_not_found` is only an error if there are no other errors since it is possible for uploading artifacts to not find a file, but not be a relevant error.